### PR TITLE
Clarify error messages

### LINKED ISSUE #230: Cryptic error message during densification of array when doing elemwise computation.
```py
>>> X = np.arange(24)
>>> zeros = sparse.zeros(X.shape)
>>> ones = sparse.ones(X.shape)
>>> sparse.where(X < 6, zeros, ones)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "./sparse/coo/common.py", line 777, in where
    return elemwise(np.where, condition, x, y)
  File "./sparse/coo/umath.py", line 47, in elemwise
    return _Elemwise(func, *args, **kwargs).get_result()
  File "./sparse/coo/umath.py", line 422, in __init__
    self._get_fill_value()
  File "./sparse/coo/umath.py", line 484, in _get_fill_value
    raise ValueError('Inconsistent fill-values in the result array: operating on the ndarray with'
ValueError: Inconsistent fill-values in the result array: operating on the ndarray with fill-values produces inconsistent results.
```

It works if the input array is sparse.

```py
>>> sparse.where(sparse.COO.from_numpy(X) < 6, zeros, ones)
<COO: shape=(24,), dtype=float64, nnz=18, fill_value=0.0>
```

### LINKED ISSUE #231: Cryptic error message during a large broadcast.
```py
>>> np.array([[1, 2], [3, 4]])*sparse.COO.from_numpy([1, 1])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/aaronmeurer/anaconda3/lib/python3.6/site-packages/sparse/coo/core.py", line 1441, in __array_ufunc__
    result = elemwise(ufunc, *inputs, **kwargs)
  File "/Users/aaronmeurer/anaconda3/lib/python3.6/site-packages/sparse/coo/umath.py", line 47, in elemwise
    return _Elemwise(func, *args, **kwargs).get_result()
  File "/Users/aaronmeurer/anaconda3/lib/python3.6/site-packages/sparse/coo/umath.py", line 423, in __init__
    self._check_broadcast()
  File "/Users/aaronmeurer/anaconda3/lib/python3.6/site-packages/sparse/coo/umath.py", line 511, in _check_broadcast
    .format(non_ndarray_shape))
ValueError: All ndarrays must be broadcastable to the shape without ndarrays (2,)
>>> np.array([[1, 2], [3, 4]])*np.array([1, 1])
array([[1, 2],
       [3, 4]])
```

### PR DESCRIPTION
**Pull request by hameerabbasi**
Fixes #231 
Fixes #230 

@asmeurer Please make sure that these error messages are clear to you... I'm used to speaking deep NumPy-speak, and so I may be a bit blinded. 🙂

### CODE REVIEW

File: sparse/coo/umath.py [hunk 1]
```diff
@@ -481,8 +481,8 @@ def _get_fill_value(self):
             fill_value = self.func(*zero_args, **self.kwargs)[()]
 
         if not equivalent(fill_value, fill_value_array).all():
-            raise ValueError('Inconsistent fill-values in the result array: operating on the ndarray with'
-                             ' fill-values produces inconsistent results.')
+            raise ValueError('Performing a mixed sparse-dense operation that would result in a dense array. '
+                             'Please make sure that func(sparse_fill_values, ndarrays) is a constant array.')
```
Comment (asmeurer) [inline sparse/coo/umath.py]: What is func?

File: sparse/coo/umath.py [hunk 2]
```diff
@@ -507,8 +507,8 @@ def _check_broadcast(self):
         )
 
         if full_shape != non_ndarray_shape:
-            raise ValueError('All ndarrays must be broadcastable to the shape without ndarrays {}'
-                             .format(non_ndarray_shape))
+            raise ValueError('Please make sure that the broadcast shape of just the sparse arrays is '
```
Comment (asmeurer) [inline sparse/coo/umath.py]: So in the case of a binary operation between dense and sparse as in the issue, "just the sparse arrays" is the one array, so broadcasting the sparse array is not allowed?

File: sparse/coo/umath.py (hunk 1 above)
Comment (hameerabbasi) [reply sparse/coo/umath.py]: The element-wise function one wants to apply.

File: sparse/coo/umath.py (hunk 2 above)
Comment (hameerabbasi) [reply sparse/coo/umath.py]: Correct.

**Comment by hameerabbasi** [comment]
@asmeurer Do you think these error messages are fine or do they need refining? Feel free to suggest other (clearer) wordings. 🙂

**Pull request merged**